### PR TITLE
LPS-178843 Migrate utils to TypeScript, Part II

### PR DESCRIPTION
**Pull request by p2kmgcl**
Second part of https://github.com/liferay-tango/liferay-portal/pull/3330
We will wait until we finish the revert process to consider sending this.

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by p2kmgcl** [comment]
@claraizquierdo ready for review! :smile:

**Comment by p2kmgcl** [comment]
ci:test:sf

**Comment by p2kmgcl** [comment]
ci:test:relevant

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/types/Criteria.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {DateValue} from "./Date";
+
+export interface CriteriaItem {
+	assetId: string,
```
Comment (claraizquierdo) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/types/Criteria.d.ts]: I think that `assetId`, `day` and `operatorNot` fields are not present in the criteria items.

**Review by claraizquierdo — CHANGES_REQUESTED** [review]
Hi @p2kmgcl ,
everything seems to be working fine. I just left two comments :)

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/utils.ts [hunk 2]
```diff
@@ -46,31 +44,29 @@ export function generateGroupId() {
...
- * @param {string} plural The language key in plural form.
- * @param {number} count The amount to display in the message.
- * @param {boolean} toString If the message should be converted to a string.
- * @return {(string|Array)} The translated message.
  */
-export function getPluralMessage(singular, plural, count = 0, toString) {
+export function getPluralMessage(
+	singular: string,
+	plural: string,
+	count = 0,
```
[the remark this answers is not in the document]
Comment (p2kmgcl) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/utils.ts:51]: Done! :smile:

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/types/Criteria.d.ts (hunk 1 above)
Comment (p2kmgcl) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/types/Criteria.d.ts]: Purrfect!

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by liulukiwi** [comment]
ci:forward:force

**Comment by victorg1991** [comment]
ci:forward:force

**Pull request closed without merging**